### DISCUSSION TITLE
intermittent Failures when Downloading Private Actions Reposotories

### DISCUSSION THREAD

**Discussion by Bhutania** [community/community#109398] [Actions]
### Select Topic Area

Bug

### Body

Hello,

My organization has a private repository where we have our custom actions for usage. We have noticed an uptick in failures when setting up jobs and retrieving these actions with the error:
```
Download action repository:...
Error: An action could not be found at the URI 'https://api.github.com/repos/...'
```

I have verified that the action exists at the provided SHA and that this job has succeeded in being setup in the past. I have also confirmed that there are no code changes that would affect the workflow's execution during the failing run.

As this issue is intermittent (2-3 times in the last 50 runs), it is difficult to diagnose what the issue is.

Any guidance would be appreciated,
Thanks,
Arnav

**Comment by bhoehl** [comment]
We are facing the same issue.
It happened on a push to a branch.
This workflow then is using a workflow_call, which then sets up fetching the latest tarball.
In my opionion this has to do with insufficient rights for the current GITHUB_TOKEN.
It happens pretty often meanwhile.

Setup Job

Download action repository '<git url> ' (SHA:...)
Error: An action could not be found at the URI

The tarball is present at that url and sha, but as mentioned the rights to download are not appropriate.
We can download that tar ball using a personal access token.

**Comment by TaxBusby** [comment]
We've hit a handful of cases of this at our organization over the past few days. In both self-hosted and github-hosted runners. Only intermittent, and the same action / version / tarball is available in other actions successfully. There are no correlating commits or tagging changes to these repositories.

Seems like a reliability issue on GitHub's end.
